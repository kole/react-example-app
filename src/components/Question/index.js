import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import DOMPurify from 'dompurify';

const Question = styled.div`
    line-height: 1.3em;
`;

const QuestionComponent = (props) => {
    const { questionText } = props;

    // Lets clean the encoded HTML before we unsafely pass it to the DOM
    const cleanHTML = DOMPurify.sanitize(questionText);

    return <Question dangerouslySetInnerHTML={{__html: cleanHTML}}></Question>;
}

QuestionComponent.propTypes = {
    questionText: PropTypes.string.isRequired
}

export default QuestionComponent;
