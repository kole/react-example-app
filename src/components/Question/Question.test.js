/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import Question from './index.js';

describe('Question', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<Question questionText="Test question" />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
