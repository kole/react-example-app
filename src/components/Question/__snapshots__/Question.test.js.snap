// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Snapshot 1`] = `
<div
  className="sc-bdVaJa UhWAf"
  dangerouslySetInnerHTML={
    Object {
      "__html": "Test question",
    }
  }
/>
`;
